single_tech_samples/storage/sample1_data_lifecycle/README.md

```diff
@@ -0,0 +1,224 @@
+# Storage Life Cycle Managment <!-- omit in toc -->
+
+## Contents <!-- omit in toc -->
+
+- [1. Solution Overview](#1-solution-overview)
+  - [1.1. Scope](#11-scope)
+  - [1.2. Use Case](#12-use-case)
+  - [1.3. Technologies used](#14-technologies-used)
+- [2. How to use this sample](#2-how-to-use-this-sample)
+  - [2.1. Prerequisites](#21-prerequisites)
+    - [2.1.1 Software Prerequisites](#211-software-prerequisites)
+  - [2.2. Setup and deployment](#22-setup-and-deployment)
+  - [2.3. Deployed Resources](#23-deployed-resources)
+  - [2.4. Deployment validation and Execution](#24-deployment-validation-and-execution)
+  - [2.5. Clean-up](#25-clean-up)
+- [3. Troubleshooting](#3-troubleshooting)
+- [4. Notes](#4-notes)
+- [5. References](#5-references)
+
+## 1. Solution Overview
+
+The purpose of this solution is to demonstrate the creation of Azure storage accounts/containers and configuring the data lifecycle rules at scale using Terraform based IaC templates.
+
+The solution takes details about the storage accounts, containers and lifecycle rules as JSON configuration and apply it in a loop which makes it flexible, extensible and scalable to accommodate any number of storage accounts, containers and data life cycle rules.
+
+### 1.1. Scope
+
+The following items will be provisioned as a part of this sample setup:
+
+- Azure Storage Accounts.
+- Storage Containers.
+- Data lifecycle rules.
+
+Details about [how to use this sample](#2-how-to-use-this-sample) can be found in the later sections of this document.
+
+### 1.2. Use Case
+
+It is desirable to reduce the costs by optimizing the resource usage costs in cloud environments.
+
+Data lifecycle is one such feature on Azure using which you can optimize costs by moving your data to lower cost tiers when their access is less frequent. It also supports deleting obselete data to save costs.
+
+More details can be found in [this link](https://learn.microsoft.com/azure/storage/blobs/lifecycle-management-overview)
+
+We can define data lifecycle rules to move data from one tier to other. The current solution provides Infra As Code script written in Terraform, to provision these rules in Azure.
```

Comment:
Avoid 1st person (We), suggest rewrite in 3rd person passive.

Reply:
Done